src/v/cloud_storage/cache_service.cc

```diff
@@ -651,6 +656,20 @@ ss::future<cache::trim_result> cache::trim_fast(
         }
     }
 
+    auto max_carryover_files = config::shard_local_cfg()
+                                 .cloud_storage_cache_trim_carryover.value()
+                                 .value_or(0);
+    fragmented_vector<file_list_item> tmp;
```

Comment:
Would reserving some memory for `tmp` here be beneficial?
Something like `tmp.reserve(std::min(max_carryover_files, candidates.size() - candidate_i))`?